components/camel-graphql/src/main/java/org/apache/camel/component/graphql/GraphqlEndpoint.java

```diff
@@ -0,0 +1,119 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.graphql;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+import org.apache.camel.Component;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+import org.apache.camel.util.IOHelper;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+@UriEndpoint(firstVersion = "3.0.0", scheme = "graphql", title = "GraphQL", syntax = "graphql:httpUri", label = "graphql")
+public class GraphqlEndpoint extends DefaultEndpoint {
+
+    private final CloseableHttpClient httpclient;
+    @UriPath
+    @Metadata(required = true)
+    private URI httpUri;
+    @UriParam
+    private String query;
+    @UriParam
+    private String queryFile;
+    @UriParam
+    private String operationName;
+    @UriParam(prefix = "variable.", multiValue = true)
+    private Map<String, Object> variables;
+
+    public GraphqlEndpoint(String uri, Component component) {
+        super(uri, component);
+        this.httpclient = HttpClients.createDefault();
+    }
+
+    @Override
+    public Producer createProducer() throws Exception {
+        return new GraphqlProducer(this);
+    }
+
+    @Override
+    public Consumer createConsumer(Processor processor) throws Exception {
+        throw new UnsupportedOperationException("You cannot receive messages at this endpoint: " + getEndpointUri());
+    }
+
+    public CloseableHttpClient getHttpclient() {
+        return httpclient;
+    }
+
+    public URI getHttpUri() {
+        return httpUri;
+    }
+
+    public void setHttpUri(URI httpUri) {
+        this.httpUri = httpUri;
+    }
+
+    public String getQuery() {
+        if (query == null && queryFile != null) {
+            try {
+                query = IOHelper.loadText(ObjectHelper.loadResourceAsStream(queryFile));
+            } catch (IOException e) {
+                throw new RuntimeCamelException("Failed to read query file: " + queryFile, e);
+            }
+        }
+        return query;
+    }
+
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    public String getQueryFile() {
```

Comment:
@akieling you need to add javadoc to this setter/getter methods, take a look at other components. The javadoc will be used as description in the component adoc documentation file.